crates/ty_python_semantic/src/semantic_index/use_def.rs

```diff
@@ -590,6 +590,46 @@ impl<'db> UseDefMap<'db> {
             .map(|symbol_id| (symbol_id, self.end_of_scope_symbol_bindings(symbol_id)))
     }
 
+    pub(crate) fn all_reachable_symbols_with_definitions<'map>(
+        &'map self,
+        db: &'db dyn crate::Db,
+    ) -> impl Iterator<Item = (ScopedSymbolId, Type<'db>, Definition<'db>)> {
+        self.reachable_definitions_by_symbol
+            .iter_enumerated()
+            .flat_map(|(symbol_id, reachable_definition)| {
```

Comment:
nit: the second element here is a `&ReachableDefinitions`, which stores multiple bindings and declarations -- so it seems confusing to name the variable `reachable_definition` (singular)
```suggestion
            .flat_map(|(symbol_id, reachable_definitions)| {
```